### DISCUSSION TITLE
Query products that have discount

### DISCUSSION THREAD

**Discussion by EL-MEHDI-ESSAADI** [saleor/saleor#13443] [Q&A] [answered]
Is there is a way to query only products that have discount using the products query?

**Comment by peteror** [comment, marked as answer]
That would be tricky, as discounts / vouchers are set on the channel level, so the same product may have a discount in one and not in another. 
However, you could get the list of products by querying the vouchers / sales (but this needs extra permissions, so you cannot run this from a client app without hacking Saleor core a bit) like this: 

```

``` 

```query {
  vouchers(first: 100) {
    edges {
      node {
        channelListings {
          channel {
            name
          }
          discountValue
        }
        products(first:100) {edges{node{id}}}
      }
    }
  }
}
```

of course, if your promotion is defined by category / collection, instead of the products subquery, you'll need to query the category or collection, and _then_ the products: 


```query {
  vouchers(first: 100) {
    edges {
      node {
        channelListings {
          channel {
            name
          }
          discountValue
        }
        categories(first: 100) {
          edges {
            node {
              products(first:100) {edges{node{id}}}
            }
          }
        }
      }
    }
  }
}
```
My guess is that it's a very expensive / slow query though, as it can get quite recursive. 

Depending on your use case, I'd go with some combination of these.

**Comment by EL-MEHDI-ESSAADI** [comment]
In my case I changed the vouchers to sales and it worked well, thank you @peteror
